From c1bed348be43011ca1567dd53760f2ffbe6e5046 Mon Sep 17 00:00:00 2001
From: robinz1x <robinx.zhang@intel.com>
Date: Thu, 3 May 2018 17:14:23 +0800
Subject: [PATCH 20/21] Get the correct compressed size

The MtpObjectInfo#getCompressedSize() returns the value in
Java int (signed 32-bit integer), so when the MtpObject size is
larger than 2^31-1, use getCompressedSize() will encounter
the IllegalStateException error.

Use getCompressedSizeLong() instead of getCompressedSize() can
get the correct size.

Change-Id: I735c98247f65f85aca602551623b8bcda4d8c41c
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-62404
Signed-off-by: robinz1x <robinx.zhang@intel.com>
---
 packages/MtpDocumentsProvider/src/com/android/mtp/MtpManager.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/MtpDocumentsProvider/src/com/android/mtp/MtpManager.java b/packages/MtpDocumentsProvider/src/com/android/mtp/MtpManager.java
index 8f254e9..a7de631 100644
--- a/packages/MtpDocumentsProvider/src/com/android/mtp/MtpManager.java
+++ b/packages/MtpDocumentsProvider/src/com/android/mtp/MtpManager.java
@@ -200,7 +200,7 @@ class MtpManager {
             }
             if (objectInfo.getFormat() != MtpConstants.FORMAT_ASSOCIATION) {
                 if (!device.sendObject(sendObjectInfoResult.getObjectHandle(),
-                        sendObjectInfoResult.getCompressedSize(), source)) {
+                        sendObjectInfoResult.getCompressedSizeLong(), source)) {
                     throw new IOException("Failed to send contents of a document");
                 }
             }
-- 
1.9.1

